src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -88,13 +83,14 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
     }
   }
 
-  private def extractFee(tx: ErgoTransaction): Long = {
-    val propositionBytes = settings.chainSettings.monetary.feePropositionBytes
+  private def extractFee(tx: ErgoTransaction): Long =
     ErgoState.boxChanges(Seq(tx))._2
       .filter(_.ergoTree == settings.chainSettings.monetary.feeProposition)
       .map(_.value)
       .sum
-  }
+
+  private def randIndexes(qty: Int, max: Int): Seq[Int] =
+    (0 to qty).map(_ => Random.nextInt(max))
```

Comment:
The result may have duplicates. 
You probably may need to keep track of previously chosen indexes.
Or maybe google an algorithm for this task, it may have nice and efficient solution.